wallet-new/src/Cardano/Wallet/API/V1/Handlers/Transactions.hs

```diff
@@ -108,3 +110,40 @@ estimateFees aw payment@Payment{..} = do
     case res of
          Left err  -> throwM err
          Right fee -> return $ single (EstimatedFees (V1 fee))
+
+redeemAda :: ActiveWalletLayer IO
+          -> Maybe WalletId
+          -> Maybe AccountIndex
+          -> Redemption
+          -> Handler (WalletResponse Transaction)
+redeemAda layer mWId mAccIx redemption = do
```

Comment:
We can do better here -- if you replace your API types from using `QueryParam` to `CaptureWalletId` and `CaptureAccountId`, we can avoid the `Maybe` altogether.

Reply:
I didn't write the endpoint, just went with the endpoint as specified by @KtorZ . I can change it of course but that didn't feel like it was my decision to make :)

Reply:
Ok, @adinapoli-iohk insists that `CaptureWalletId` and co are the way to go here so will change, despite it being inconsistent with the other endpoint :)

Reply:
I wouldn't mind having this endpoint placed under the `Transactions` group built the following way:

`/wallets/{walletId}/accounts/{accountIndex}/certificates`

(basically, as it was before, under the accounts API, but here under the transaction API). 

In essence, it's a transaction, but still a very specific one related to one wallet and one account of that particular wallet.

Reply:
I hear you, but at the same time if we do down this line of reasoning we would have to move the entire transaction machinery under `/accounts`, because you never generate transactions out of thin air, they are always originated from a source wallet and account, so I'm not sure I buy into the dream here 😁 

To me, we should try to follow the RESTful principles (which I know some times are more art than science) -- we are manipulating transaction resources at the end of the day, because it's what we _create_ and _list_ , so that seems to suggest we should stick with the `/transactions` hierarchy here.

Reply:
Yeah, let's get back to the principles here. We are creating a `transactions`, right, so we ought to do a `POST /transactions`. But, it's actually a kind of special transaction, it's a certificate redemption. Thus, I see three options here:

- We consider certificates as a resource on its own and create a new set of handlers under `/certificates`. However, their representation is strictly identical to transactions, they don't have any representation on their own. So this feels weird.

- We consider certificates as a sub-resource of a Transaction which also feels weird. It's just a special kind of transactions. So having something like `/transactions/certificates` feels really RPC-like. Going this way, let's be honest and simply name this `/redemption-transactions` ... Meh.

- A third option would be to simply extend the current `POST /transactions` endpoint to accept either a `Payment | Redemption`, picking the right handler depending on which inputs are received. Extending the `Redemption` type to:

```hs
data Redemption = Redemption
    { redemptionRedemptionCode   :: ShieldedRedemptionCode
    , redemptionMnemonic         :: Maybe RedemptionMnemonic
    , redemptionSpendingPassword :: SpendingPassword
    , redemptionTarget           :: PaymentSource
    } deriving (Eq, Show, Generic)
```

Even though `PaymentSource` feels like a wrong name now :upside_down_face: as it is more like a `FullyQualifiedAccount` but well, you know what we say about naming things ^^"

What's your opinion on that @adinapoli-iohk ?